cds_ils/importer/views.py

```diff
@@ -76,15 +82,28 @@ def __init__(self, serializers=None, *args, **kwargs):
     @need_permissions("document-importer")
     def get(self, log_id, offset):
         """Returns the detail views of each subtask by given offset."""
-        log = None
         try:
             log = db.session.query(ImporterImportLog).get(log_id)
-        except ObjectDeletedError:
-            abort(404)
-        if log:
             return self.make_response(log, record_offset=offset)
-        else:
-            abort(404, "Task not found")
+        except ObjectDeletedError:
+            abort(404, "The task log was deleted.")
+        except NoResultFound:
+            abort(404, "Task does not exist.")
+
+    def post(self, log_id):
+        """Cancels the ongoing celery task."""
+        try:
+            log = db.session.query(ImporterImportLog).get(log_id)
+            celery_task = AsyncResult(log.celery_task_id)
+            log.status = ImporterTaskStatus.CANCELLED
+            db.session.commit()
+            if not celery_task.ready():
+                celery_task.revoke()
```

Comment:
I guess it is a nice to have? Isn't it better to let it exist on itself so that it will call `log.finalize`? It might revoked it before calling that...